MSAL/test/unit/MSALB2CPolicyTests.m

```diff
@@ -0,0 +1,148 @@
+    
+    application.tokenCache = cacheUtil.cache;
+    
+    XCTAssertNotNil(application);
+    XCTAssertNil(error);
+    
+    __block dispatch_semaphore_t dsem = dispatch_semaphore_create(0);
+    
+    [application acquireTokenForScopes:@[@"fakeb2cscopes"]
+                       completionBlock:^(MSALResult *result, NSError *error)
+     {
+         XCTAssertNil(error);
+         XCTAssertNotNil(result);
+         
+         NSString *userIdentifier = [NSString stringWithFormat:@"1-b2c_1_policy.%@", [MSALTestIdTokenUtil defaultTenantId]];
+         XCTAssertEqualObjects(result.user.userIdentifier, userIdentifier);
+         dispatch_semaphore_signal(dsem);
+     }];
+    
+    while (dispatch_semaphore_wait(dsem, DISPATCH_TIME_NOW))
+    {
+        [[NSRunLoop mainRunLoop] runMode:NSDefaultRunLoopMode beforeDate: [NSDate distantFuture]];
+    }
+    
+    // Now acquiretoken call with second policy (b2c_2_policy)
+    NSString *secondAuthority = @"https://login.microsoftonline.com/tfp/contosob2c/b2c_2_policy";
+    
+    // Override oidc and token responses for the second policy
+    oidcResponse = [MSALTestURLResponse oidcResponseForAuthority:secondAuthority
+                                                     responseUrl:@"https://login.microsoftonline.com/contosob2c"
+                                                           query:@"p=b2c_2_policy"];
+    
+    tokenResponse = [MSALTestURLResponse authCodeResponse:@"i am an auth code"
+                                                authority:@"https://login.microsoftonline.com/contosob2c"
+                                                    query:@"p=b2c_2_policy"
+                                                   scopes:[NSOrderedSet orderedSetWithArray:@[@"fakeb2cscopes", @"openid", @"profile", @"offline_access"]]
+                                               clientInfo:@{ @"uid" : @"1-b2c_2_policy", @"utid" : [MSALTestIdTokenUtil defaultTenantId]}];
+    
+    [MSALTestURLSession addResponses:@[oidcResponse, tokenResponse]];
+    
+    // Use an authority with a different policy in the second acquiretoken call
+    [application acquireTokenForScopes:@[@"fakeb2cscopes"]
+                  extraScopesToConsent:nil
+                             loginHint:nil
+                            uiBehavior:MSALUIBehaviorDefault
+                  extraQueryParameters:nil
+                             authority:secondAuthority
+                         correlationId:nil
+                       completionBlock:^(MSALResult *result, NSError *error) {
+                           
+                           XCTAssertNil(error);
+                           XCTAssertNotNil(result);
+                           
+                           NSString *userIdentifier = [NSString stringWithFormat:@"1-b2c_2_policy.%@", [MSALTestIdTokenUtil defaultTenantId]];
+                           XCTAssertEqualObjects(result.user.userIdentifier, userIdentifier);
```

Comment:
We're making sure that the returned userIdentifier is as expected. But it's not an essential part.